lib/main.php

```diff
@@ -482,6 +482,10 @@ function convert ($body, $query_string, $set_properties, $is_skip_variations = f
           if (!isset($variation['sku']) && isset($body['sku'])) {
             $variation['sku'] = $body['sku'] . '-' . rand(100, 999);
           }
+          if (!isset($variation['description']) && isset($body['body_html']) && $has_variant_description) {
+            $body_text_variation = str_replace('&nbsp;', ' ', preg_replace('#<[^>]+>#', ' ', $body['body_html']));
+            $variation['description'] = utf8_sanitize($body_text_variation);
+          }
           $variation = array_merge($body, $variation);
```

Comment:
Tem esse `merge` e chama o `convert()` depois, quer dizer, quando chama o `$variation['body_html']` deve existir e daí ele já estaria chegando em https://github.com/ecomplus/products-feed/blob/66b0c4bca96a1e8ffa8ebd1b863074c1642b58da/lib/main.php#L170

Se isso não está acontecendo deve ser um bug, mas não vejo porque esse arg `$has_variant_description` adicional nem vejo como setar `$variation['description']` ajudaria.
Talvez eu tenha entendido errado o que você quer fazer, repare que `$variation` não é uma row do XML.

Reply:
Agora entendi aqui o código, ele não é uma row. De toda forma, ele pega cada variation na função convert e cria uma nova entry. Vi que tem um array_merge($body, $variation), em teoria ele deveria juntar toda a informação presente do body e com o variation para um novo variation, nesse caso já com a description e não está acontecendo, vou ver se acho algo mais

Reply:
@leomp12 mudei aqui pelo que achei https://stackoverflow.com/questions/455700/what-is-the-best-method-to-merge-two-php-objects, possivelmente o outro array_merge não deve estar funcionando também

Reply:
Claro que tá funcionando, se não estivesse não teria a descrição em variação nenhuma, de loja nenhuma, e tem, até te mostrei no issue.
Não é aí o BO não...